lambdas/batch_process/batch_process_lambda.py

```diff
@@ -167,15 +167,15 @@ def batch_proc_once():
         except:
             print("Temporal parameter not found in batch proc. Defaulting to false.")
             temporal = False
+
         try:
-            if p.historical is True:
-                historical = True
+            processing_mode = p.processing_mode
+            if p.processing_mode == "historical":
```

Comment:
consider using the variable you created on the line above (`if processing_mode`). Unless you really want to highlight the use of `p.some_attrib`

Reply:
It's a bit confusing but both variables are needed. processing_mode is what's being used for the rest of the logic. The reason p.xxx is not being used for that is because it may not exist if the underlying json doesn't contain it. So instead of using p.xxx throughout the code and catching exception of it not existing, I chose to use another variable that can be assigned a default value